uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Service/CiviHRStatsGatherer.php

```diff
@@ -51,6 +62,11 @@ public function gather() {
       $stats->addReportConfigurationAgeGroup($group);
     }
 
+    foreach ($this->roleService->getRoleNames() as $rid => $roleName) {
```

Comment:
Would it improve performance if we had the `getLatestLoginByRole` function, rather than get latest login for each role passed in, It will internally get the last login for all the roles using a single query. 
We can then loop through the returned results to `setMostRecentLoginForRole` for each role.

Reply:
Performance is a good point, and I'll change this. I did really like having the simpler function definition of accepting a string and returning `\DateTime | null`, whereas now it returns an array, where the indices are the role names. It does surely improve performance, but in general I think I prefer having easy to understand functions where you don't need to understand the structure of the return type to use it.